### ISSUE TITLE
Feature request.  Need ability to change font / size in menus, dialogs, etc.

### ISSUE THREAD

**Issue by jeff-bowles** [notepad-plus-plus/notepad-plus-plus#5648]
### Feature request - Application itself needs a user-selectable font
The text of the menus and dialog boxes needs to be larger for the gray-haired crowd.

### Steps to Reproduce the Issue
The example that prompted my feature request:  Plugins --> Plugins Admin.  The descriptions of the plugins are hard to read.

### Expected Behavior
Something like Settings --> General --> Notepad++ Font

**Comment by MetaChuh** [comment]
@jeff-bowles 

custom font sizes, for other areas than the text editor area, would be too difficult to implement, that's why it will eventually never been implemented.
but usually the font size is quite readable for us gray-haired people, as long as we have a good display:

this is what notepad++ looks like on a 4k retina display with 200% scaling:
(please click on the picture for a full size display)

<img width="1028" alt="" src="https://user-images.githubusercontent.com/45773305/57650694-ac3cfe00-75cb-11e9-9ed8-e22cf83853fd.png">

and this is what it looks like on the same display with system scaler enabled and still 200% scaling.
<img width="1030" alt="" src="https://user-images.githubusercontent.com/45773305/57650911-3e450680-75cc-11e9-9ebd-473406cf883e.png">

is it the same on your screen, or is your display more difficult to read than the screenshots above ?

if your screen is more difficult to read, what is your resolution, display size in inches, and scaling factor you have currently set at your windows display settings ?
please also provide us with your debug information, by going to the notepad++ menu `? > debug info... > copy debug info into clipboard`.

best regards.

-----
```
general notes:

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).
please file an issue as easy to understand and reproducible as possible.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
(no extra account is needed, just use your github account to sign in)
```
[<p align="center"><kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd></p>](https://notepad-plus-plus.org/community/)

**Comment by jeff-bowles** [comment]
My display is 1920 x 1080 on a 15" laptop, so the 8pt characters in the NPP dialog boxes are pretty small.
Perhaps I'll have a go at the source code myself.  I really don't expect there to be much work to change the fonts from 8pt to 10pt or 12pt.

**Comment by OneOfMany07** [comment]
OS font settings should be respected for application elements.  I haven't played around with much, but my guess is they're ignored currently.  Maybe would require a rewrite as a different app type (Universal or whatever they call the new hotness).

**Comment by ortollj** [comment]
Hi
I can't read the notepad++ menu on my Ubuntu 18.04, output on  my tv screen using HDMI.
this is what I see:
![notepadMenuPB](https://user-images.githubusercontent.com/5411158/81465667-d7046700-91cb-11ea-8d02-39430c78f5f9.png)

**Comment by OneOfMany07** [comment]
Probably should file a bug of your own.  Unless you're giving a reason this bug should be fixed.

**Comment by sasumner** [comment]
@ortollj 
Ubuntu isn't an officially supported platform.
Thus, @OneOfMany07 's advice to "open a bug of your own" probably isn't going to get very far.

**Comment by brandpeter** [comment]
On windows10 one can set the system font in the registry (as shown in the attached screenshot). I selected Gadugi which has a visible difference between lowercase L and uppercase I. This is important when working with CamelCase filenames which will end up on case-sensitive Unix systems. For example the file in the screenshot ColIsvisible. It appears with the correct font in the Explorer panel, but not in the Tab or rename dialog.  NPPs menu does use the correct font. NPP should therefore be consistent in its use of system font.
![Screenshot 2020-11-15 122858](https://user-images.githubusercontent.com/4787029/99182577-5c9a3e00-273e-11eb-920b-804b740e0a26.png)

**Comment by Ioan-Popa** [comment]
This is a very important issue.

I have a 27" WQHD monitor running at a resolution of 2560x1440. Windows should be scaling everything up to 125%. Unfortunately the Notepad++ menu font and toolbar icons remain disproportionately small.

Although I have been using Notepad++ for a long time for quick and uncomplicated edits to text files, I finally decided to replace it with Microsoft Visual Studio Code, since it does not manifest the above mentioned problem. The only drawback seems to be the slightly longer time it takes to start.

**Comment by PhillipBSDL** [comment]
Yes this is a problem, as screen resolution increases, the Notepad++ font size becomes ever increasingly smaller and more difficult to use with each passing year

**Comment by luapfr** [comment]
Hi, I would also like this feature very much.

**Comment by sophware** [comment]
I used the Windows method ("make things bigger" in settings) to improve the font situation, which was making it impossible for me to use the menu. It worked, but the icons are still tiny.

**Comment by ArgyleTrees** [comment]
ricght-click on the notead++.exe file.  change the compatibilty settings to make it compatible with Windows 8. That will fix the scaling issue.

**Comment by ArgyleTrees** [comment]
right-click on the notepad++.exe, select properties --> change the compatibility setting. --> chnage how the High DPI will be processed --> Use the System Enhance DPI processing. 

<img width="301" alt="image" src="https://user-images.githubusercontent.com/113072765/213899038-be839db2-2b41-4dbb-9052-10c92ba39e62.png">

**Comment by exazettabit** [comment]
Hey, I would like this feature as well. 

As the current application font of Notepad++ is consistent and stable across all menus (toolbar, settings menu, etc) it seems that all we need to do is replace the central value that is assigned to the font-size inside the program.

Maybe we can have prearranged sizes like standard - middle - big?
These sizes would need to be tested beforehand and word wrapped as well as the menu size should be increased in some places to accommodate the bigger text. 

So changing the font-size value itself would not be as difficult as replacing the icons (for which a whole new set of icons had to be created last year).
But the menu needs to fit around the words, so "fitting" it inside the GUI will take time and manual testing for the developers.

**Comment by JasonMendoza2008** [comment]
Unfortunately, the menu is currently unreadable for me too.

**Comment by NathanSweet** [comment]
I just want the tab font larger. I can live with small menus, but not tabs.

**Comment by sjscheider** [comment]
To make the tab font larger, make sure "Reduce" is not checked in Settings -> Preferences -> General.

**Comment by Ioan-Popa** [comment]
Thank you. This helped.

**Comment by exazettabit** [comment]
That is actually a lot better! Hopefully they will implement font size possibilities for the rest of the program too some time in the future.

**Comment by JasonMendoza2008** [comment]
@sjscheider This was already not checked.

**Comment by nkilm** [comment]
is there any fix for this? I'm using Notepad++ on Ubuntu 24 with the help of [wine](https://askubuntu.com/questions/15518/how-do-i-install-wine). 

![Image](https://github.com/user-attachments/assets/c2857b7f-6345-45c9-85be-be079196dc98)

**Comment by JasonMendoza2008** [comment]
> is there any fix for this? I'm using Notepad++ on Ubuntu 24 with the help of [wine](https://askubuntu.com/questions/15518/how-do-i-install-wine).
> 
> ![Image](https://github.com/user-attachments/assets/c2857b7f-6345-45c9-85be-be079196dc98)

No there is no fix at the moment.

<img width="777" alt="Image" src="https://github.com/user-attachments/assets/965b2824-6802-4a22-b29e-c19b01e73362" />

**Comment by nkilm** [comment]
> > is there any fix for this? I'm using Notepad++ on Ubuntu 24 with the help of [wine](https://askubuntu.com/questions/15518/how-do-i-install-wine).
> > ![Image](https://github.com/user-attachments/assets/c2857b7f-6345-45c9-85be-be079196dc98)
> 
> No there is no fix at the moment.
> 
> <img alt="Image" width="777" src="https://private-user-images.githubusercontent.com/56092290/450834552-965b2824-6802-4a22-b29e-c19b01e73362.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg5NjA1ODksIm5iZiI6MTc0ODk2MDI4OSwicGF0aCI6Ii81NjA5MjI5MC80NTA4MzQ1NTItOTY1YjI4MjQtNjgwMi00YTIyLWIyOWUtYzE5YjAxZTczMzYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjAzVDE0MTgwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUxM2ZhNjk2ODE3ZWM1ZWU5Nzg4MDE1ZmZmNDZkYTAxOWE3MThkM2JiMjE2NmQyNDE0Njg4OTdiMjYxZDcxMmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aBQ8ZIhMHnf9HsaMVSrBvr909FpBBuysPrFTEkJtp_w">

oh that's sad to hear.
